pkg/backend/display/jsonmessage.go

```diff
@@ -77,6 +82,27 @@ func cursorDown(out io.Writer, ti termInfo, l int) {
 	}
 }
 
+// Progress describes a message we want to show in the display.  There are two types of messages,
+// simple 'Messages' which just get printed out as a single uninterpreted line, and 'Actions' which
+// are placed and updated in the progress-grid based on their ID.  Messages do not need an ID, while
```

Comment:
"do not need" does that mean message _can_ have an ID, or that they _never_ have an ID?